src/platform/forms-system/src/js/review/StringField.jsx

```diff
@@ -13,18 +14,69 @@ import {
  * in the config
  */
 export default function StringField(props) {
-  const { registry, schema, uiSchema, formData } = props;
-  const uiOptions = getUiOptions(uiSchema);
+  const {
+    registry = getDefaultRegistry(),
+    schema,
+    uiSchema,
+    formData,
+    children,
+    formContext,
+    id,
+  } = props;
+  const realUiSchema = checkForNestedUISchema(uiSchema);
+  const uiOptions = getUiOptions(realUiSchema);
   const labels = uiOptions.labels || {};
   const enumOptions = Array.isArray(schema.enum) && optionsList(schema);
+  const isRoot = id === 'root';
+  let Widget = _.get('ui:reviewWidget', realUiSchema);
 
-  let Widget = _.get('ui:reviewWidget', uiSchema);
   if (!Widget) {
-    const defaultWidget = schema.format || (enumOptions ? 'select' : 'text');
-    Widget = getWidget(
-      schema,
-      uiOptions.widget || defaultWidget,
-      registry.widgets,
+    if (schema.type !== 'object' && schema.type !== 'array') {
+      const defaultWidget = schema.format || (enumOptions ? 'select' : 'text');
+      Widget = getWidget(schema, defaultWidget, registry.widgets);
+    } else {
+      return children;
+    }
+  }
+
+  if (
+    (isRoot && schema.type === 'object') ||
+    (isRoot && schema.type === 'array')
+  ) {
```

Comment:
- `A` represents `isRoot`
- `B` represents `schema.type === 'object'`
- `C` represents `schema.type === 'array'`

| A | B | C | (A && B) \|\| (A && C) |
| :-: | :-: | :-: | :-: |
| T  | T  | T  | T |
| T  | T  | F  | T |
| T  | F  | T  | T |
| T  | F  | F  | F |
| F  | T  | T  | F |
| F  | T  | F  | F |
| F  | F  | T  | F |
| F  | F  | F  | F |

Compare the table for the above boolean expression to this one:

| A | B | C | A && (B \|\| C) |
| :-: | :-: | :-: | :-: |
| T  | T  | T  | T |
| T  | T  | F  | T |
| T  | F  | T  | T |
| T  | F  | F  | F |
| F  | T  | T  | F |
| F  | T  | F  | F |
| F  | F  | T  | F |
| F  | F  | F  | F |


I think I've done this right, but I might have missed a detail.  If I _have_ been correct, then you can use the simpler boolean expression.

Reply:
For the tables, I see what you mean but it wouldn't be possible for B and C to both be true as the schema type could only be one of those values. But you are correct in that we can use the  `A && (B || C)` boolean expression.